### PR TITLE
feat: Add samconfig.toml to All Init Templates

### PR DESCRIPTION
**Pull request by mildaniel**
*Issue #, if available:*
N/A

*Description of changes:*
1. Adds a default `samconfig.toml` file to all of the init templates. 
2. Adds several helper scripts to automate the process instead of copying and pasting.
3. Adds scripts to check if `samconfig.toml` exists in all templates.
4. Adds GitHub action to invoke the checking script on all PRs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: scripts/common.py
Comment (sriram-mv) [inline scripts/common.py:32]: we may need to check for symlinks ordinarily to avoid loops but we do control the source code here, so should be fine.

File: scripts/samconfig-zip.toml
Comment (sriram-mv) [inline scripts/samconfig-zip.toml:6]: Not strictly required, we do show `sam-app` as the prompt during --guided flows. Should we prefix with `sam-app-{}` to give it a feel of a namespace?

File: scripts/common.py
Comment (mildaniel) [reply scripts/common.py:51]: Good call-out. I didn't consider that when writing this but we should be okay here. Definitely something to keep in mind going forward though

File: scripts/samconfig-zip.toml
Comment (mildaniel) [reply scripts/samconfig-zip.toml]: I like it.

File: .github/workflows/check-samconfig.yml
Comment (sriram-mv) [inline .github/workflows/check-samconfig.yml:13]: What happens if this script fails? Does it block all merges? Do we need tests for it? Not blocking approval since the actions have passed.

**Review by sriram-mv — APPROVED** [review]

File: .github/workflows/check-samconfig.yml
Comment (mildaniel) [reply .github/workflows/check-samconfig.yml:13]: AFAIK, it can only be added to the required approval flow in the PR settings after merging it in. Until then, it'll show up as a failure but shouldn't block any PRs from being merged.

**Review by lucashuy — APPROVED** [review]

**Comment by mildaniel** [comment]
Based on internal discussions, we will be moving this into SAM-CLI for the sake of maintainability.

**Pull request closed without merging**
